### PR TITLE
Refactors Regenerate Organs, and a few organ helpers

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

Refactors regenerate organs to be slightly more intelligent in handling organ changes and replacements.
Noteably:
 - We don't remove organs that were modified by the owner; such as changing out your heart for a cybernetic
 - We early break out of the for loop if they aren't supposed to have an organ there and remove it
 - We check for the organ already being correct, and just healing it and continuing if it is

Also changes the names of some of the organ helpers into snake_case
### Mapping March
Ckey to receive rewards: N/A

### CODE REVIEW

**Comment by Mothblocks** [comment]
youve changed over 200 files in this pr. what files here actually matter for review?

**Comment by ZephyrTFA** [comment]
> youve changed over 200 files in this pr. what files here actually matter for review?

species.dm

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -313,103 +313,139 @@ GLOBAL_LIST_EMPTY(features_by_species)
...
+		var/should_have = new_organ.get_availability(src, organ_holder)
+
+		// Check for an existing organ, and if there is one check to see if we should remove it
+		var/health_pct = 1
+		var/remove_existing = !isnull(existing_organ)
+		if(remove_existing)
+			if(existing_organ.zone in excluded_zones)
+				remove_existing = FALSE
+			else if(existing_organ.organ_flags & ORGAN_UNREMOVABLE)
+				remove_existing = FALSE
```
Comment (Jacquerel) [inline code/modules/mob/living/carbon/human/species.dm:401]:
```suggestion
		var/remove_existing = !isnull(existing_organ) && !(existing_organ.zone in excluded_zones) && !(existing_organ.organ_flags & ORGAN_UNREMOVABLE)
```

This line does get pretty long when concatenated but unless you think it presents a seriously problem for readability it's probably better than having the rest of that block indented?

Alternately:
```suggestion
		var/keep_existing = isnull(existing_organ) || (existing_organ.zone in excluded_zones) || (existing_organ.organ_flags & ORGAN_UNREMOVABLE)
```
I find the positive version is often easier to keep in the head than the negative one
Will git freak out if I put two suggestions in one comment? I've never tried it before.

**Review by Jacquerel — CHANGES_REQUESTED** [review]
Looks clean and sensible, only got one nitpick.
I will admit I skimmed what wasn't in `species.dm` but it largely looks like just changing camel to snakecase.

File: code/modules/mob/living/carbon/human/species.dm (hunk 1 above)
Comment (ZephyrTFA) [reply code/modules/mob/living/carbon/human/species.dm:401]: For this instance I dislike how putting it all in one statement impacts readability, but at the same time anyone working with species code should understand and be able to read it

Comment (Jacquerel) [reply code/modules/mob/living/carbon/human/species.dm]: It seems that I may have tricked you with a slightly incorrect web edit.

**Pull request merged**